### PR TITLE
Allow `groups_exempted`, `groups_enforced` versions of their `gids_` counterparts

### LINKED ISSUE #46: Wishlist: resolve group names
For maintainability I'd love to be able to specify group names in addition to GIDs for gids_exempted.

My rust-foo is superficial at best but I'm thinking this should be doable by extending is_sudoing_from_exempted_gid() , what do you think?

### PR DESCRIPTION
**Pull request by stouset**
Closes #46.

Pushing the work so far, but I'll add tests when I have a moment.

### CODE REVIEW

File: sudo_pair/src/lib.rs [hunk 1]
```diff
@@ -634,11 +668,19 @@ impl<'a> From<&'a OptionMap> for PluginOptions {
...
         }
     }
 }
+
+/// Converts a POSIX `groupname` into its equivalent `gid` using `getgrnam(3)`.
+/// If the groupname cannot be found, returns `None`.
+fn groupname_to_gid(groupname: &str) -> Option<gid_t> {
+    unsafe {
+        let groupname_c = CString::new(groupname).ok()?;
+        let group       = libc::getgrnam(groupname_c.as_ptr());
```
Comment (klieth) [inline sudo_pair/src/lib.rs]: You should only need to put the unsafe around the libc::getgrnam I believe--I think it's the only unsafe thing in this function.

Comment (stouset) [reply sudo_pair/src/lib.rs]: `getgrnam` returns a `struct group *` (a pointer!) which we dereference with the unsafe method [`as_ref`](https://doc.rust-lang.org/std/primitive.pointer.html#method.as_ref). But the CString part is safe, so we can pull it out.

Comment (stouset) [reply sudo_pair/src/lib.rs]: Fixed!

**Review by klieth — DISMISSED** [review]

**Comment by stouset** [comment]
Closing this due to lack of interest from the requesting party. I'm open to updating this and merging if someone else believes it would be useful.

**Pull request closed without merging**